miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -371,6 +377,52 @@ private void validateAliasUniqueness(Sample sample) throws IOException {
     }
   }
 
+  /**
+   * Adds any external names in the sample's attached identity to the identity stored in the database
+   * 
+   * @param sample
+   * @throws IOException
+   */
+
+  private void addExternalNames(DetailedSample sample, SampleIdentity tempIdentity) throws IOException {
+    if (sample.getParent() == null) {
+      throw new IllegalArgumentException(ERR_MISSING_PARENT_ID);
+    }
+    if (tempIdentity.getExternalName() == null) return;
+
+    DetailedSample child = sample;
+    DetailedSample parent = sample.getParent();
+    while (parent.getParent() != null) {
+      child = parent;
+      parent = child.getParent();
+    }
+
+    if (!isIdentitySample(parent)) {
+      throw new IllegalStateException("Missing Identity at root of hierarchy");
+    }
+
+    SampleIdentity identity = (SampleIdentity) parent;
+    List<String> identityExternalNames = COMMA.splitAsStream(identity.getExternalName()).map(String::trim).collect(Collectors.toList());
+    List<String> tempExternalNames = COMMA.splitAsStream(tempIdentity.getExternalName()).map(String::trim).collect(Collectors.toList());
+    Set<String> lowerCaseIdentityExternalNames = identityExternalNames.stream().map(String::toLowerCase).collect(Collectors.toSet());
+    tempExternalNames.stream().forEach(name -> {
+      try {
+        if (!lowerCaseIdentityExternalNames.contains(name.toLowerCase()) && !(isUniqueExternalNameWithinProjectRequired() &&
+            (identity.getProject() == null
+                || getIdentitiesByExternalNameOrAliasAndProject(name, identity.getProject().getId(), true).size() > 0))) {
+          identityExternalNames.add(name);
+        }
+      } catch (IOException e) {
+        log.error("Failed to retrieve all external names: " + e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Format specifiers should be used instead of string concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)